nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -45,6 +45,26 @@ impl DataStore {
         }
         type TxnError = TransactionError<VolumeCreationError>;
 
+        // Grab all the targets that the volume construction request references.
+        // Do this outside the transaction, as the data inside volume doesn't
+        // change and this would simply add to the transaction time.
+        let crucible_targets = {
+            let vcr: VolumeConstructionRequest =
+                serde_json::from_str(&volume.data()).map_err(|e| {
+                    Error::internal_error(&format!(
+                        "serde_json::from_str error in volume_create: {}",
+                        e
+                    ))
+                })?;
+
+            let mut crucible_targets = CrucibleTargets::default();
+            read_only_resources_associated_with_volume(
+                &vcr,
+                &mut crucible_targets,
+            );
+            crucible_targets
+        };
+
         self.pool()
             .transaction(move |conn| {
```

Comment:
Not needed for this PR, but this could also likely be converted into a CTE, if we want.